_source/_posts/2021-09-12-spring-boot-spa.md

```diff
@@ -0,0 +1,686 @@
+
+const yourOktaUri = '${yourOktaUri}';
+const clientId = '${clientId}';
+
+const oktaSignIn = new OktaSignIn({
+    baseUrl: `${yourOktaUri}`,
+    clientId: clientId,
+    redirectUri: 'http://localhost:8080/login/callback',
+    authParams: {
+        pkce: true,
+        issuer: `${yourOktaUri}/oauth2/default`,
+        display: 'page',
+        scopes: ['openid', 'profile', 'email']
+    }
+});
+
+const oktaAuth = new OktaAuth({
+    issuer: `${yourOktaUri}/oauth2/default`,
+    clientId: clientId,
+    redirectUri: window.location.origin + '/login/callback',
+    scopes: ['openid', 'profile', 'email']
+})
+
+export { oktaAuth, oktaSignIn };
+```
+
+In the file above, **you need to replace the placeholders `${...}` with the appropriate values**. Before you can do this, however, you need to use the CLI to create the OIDC application on the Okta servers, which you will do directly below.
+
+## Create A Single-Page OIDC Application On Okta
+
+This time you're going to create a single-page application for the front-end (instead of the web application type you used for the resource server). Enter the following command into a shell.
+
+```bash
+okta apps create
+```
+
+Press `enter` to accept the default **Application name** `okta-app`. (Application name here refers to the OIDC application created on the Okta servers).
+
+For **Type of Application**, Select `2: Single Page App`.
+
+Change the **Redirect URI** to `http://localhost:8080/login/callback`.
+
+Press `enter` to accept the default **Post Logout Redirect**.
+
+You should see some output like the following.
+
+```bash
+Okta application configuration: 
+Issuer:    https://{yourOktaUri}/oauth2/default
+Client ID: {clientId}
+```
+
+The values in brackets will, of course, be your actual values. 
+
+**You need to go back to the `src/okta/index.js` file and replace the `yourOktaUri` and `clientId` variables at the top of the file with your values.** The value for `yourOktaUri` is the Okta URI without any further path specifiers. It will look like this: `https://dev-123456.okta.com`
+
+One of the things that the Okta CLI does for you is to add your application's base URL to the CORS trusted origins for the Okta auth server. This is necessary because the the Okta Sign-In Widget will be making cross0-origin requests, that, as noted above, would be blocked unless CORS is properly handled. You can see this by going to your Okta developer dashboard and selected **Security** and **API** from the left-side menu and selected **Trusted Origins** from the API tab. You'll see that (in this case) `http://localhost:8080` was added as a trusted origin of type `CORS Redirect`. 
+
+{% img blog/spring-boot-spa/trusted-origin.png alt:"Trusted origins for Okta CORS" width:"800" %}{: .center-image }
+
+You can read more about CORS and Okta in the [Okta documentation](https://developer.okta.com/docs/guides/enable-cors/overview/).
+
+## Finish the Vue Front-End Client App
```

Comment:
Can you update the example to redirect to a login page instead of an embedded log in.
There is an example here: https://github.com/okta/samples-js-vue/tree/master/okta-hosted-login

Let me know if you run into any issues

Reply:
@bdemers This post is supposed to do embedded login to replace [this post](https://developer.okta.com/blog/2017/10/27/secure-spa-spring-boot-oauth) since it uses Spring Boot 2.1 and no longer works. Related: https://github.com/oktadev/okta-blog/issues/651

Reply:
Ahh! thanks for calling that out! :D 
Sorry for the noise @moksamedia!

Reply:
So I should leave this as embedded, right?

Reply:
@moksamedia yup, keep it as embedded

Reply:
Do y'all want to do another post with the redirect instead? Would be pretty easy to knock out since I've just written this one.

Reply:
Sorry, for the delay! The short answer is yes! but we should probably space out the post a bit